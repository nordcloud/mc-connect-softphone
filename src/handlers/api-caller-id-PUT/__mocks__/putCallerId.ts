export const putCallerId = jest.fn();
