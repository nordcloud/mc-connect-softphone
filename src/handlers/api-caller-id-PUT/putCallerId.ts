import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
import { DynamoDBDocumentClient, PutCommand } from '@aws-sdk/lib-dynamodb';

import { CALLER_ID_TABLE } from '../../consts';
import { User } from '../../types';

const ddbDocClient = DynamoDBDocumentClient.from(new DynamoDBClient({}));

type Props = {
  user: User;
  callerId: string;
};

export function putCallerId({ user, callerId }: Props) {
  const command = new PutCommand({
    TableName: CALLER_ID_TABLE,
    Item: {
      username: user.email,
      callerId,
    },
  });

  return ddbDocClient.send(command);
}
