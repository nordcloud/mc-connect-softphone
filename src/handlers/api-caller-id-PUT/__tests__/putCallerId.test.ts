import { DynamoDBDocumentClient, PutCommand } from '@aws-sdk/lib-dynamodb';
import { mockClient } from 'aws-sdk-client-mock';

import { putCallerId } from '../putCallerId';

const ddbMock = mockClient(DynamoDBDocumentClient);

const args = [
  {
    user: { email: 'dummyUserEmail' },
    callerId: 'dummyCallerId',
  },
] as unknown as Parameters<typeof putCallerId>;

beforeEach(() => {
  ddbMock.reset();
  ddbMock.on(PutCommand).resolves({});
});

describe('putCallerId', () => {
  it('sends "PutCommand" with correct args', async () => {
    await putCallerId(...args);
    expect(ddbMock.commandCalls(PutCommand)[0].args[0].input).toMatchSnapshot();
  });

  it('resolves with a correct value', () => {
    return expect(putCallerId(...args)).resolves.toMatchSnapshot();
  });
});
