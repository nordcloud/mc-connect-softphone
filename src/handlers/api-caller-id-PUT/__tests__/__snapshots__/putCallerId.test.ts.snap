// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`putCallerId resolves with a correct value 1`] = `{}`;

exports[`putCallerId sends "PutCommand" with correct args 1`] = `
{
  "Item": {
    "callerId": "dummyCallerId",
    "username": "dummyUserEmail",
  },
  "TableName": "mc-connect-softphone-prod-agents-caller-id",
}
`;
